### ISSUE TITLE
ZWA039 wrong temperature (used dew value)

### ISSUE THREAD

**Issue by duck-wolv** [domoticz/domoticz#5524]
Version: 2022.2 (build 14922)
Build Hash: 338aa8e9b
Compile Date: 2023-01-06 16:48:51
dzVents Version: 3.1.8
Python Version: 3.7.3 (default, Oct 31 2022, 14:04:00) [GCC 8.3.0]

To confirm what @dwhile commented a few months ago: I do have exactly the issue which started when I moved from OpenZwave to Z-Wave JS UI and using the Aeotec ZWA039  Temperature sensor.

**Result:** Temperature shown is likely the Dew temperature and not the air temperature.

**MQTT topics are received by Domoticz:**
2023-01-06 20:46:03.893 Status: ZJS: discovered: Slaapkamer MR/Slaapkamer MR_general_purpose (unique_id: zwavejs2mqtt_0xc3631a9c_42-48-0-General_Purpose)
2023-01-06 20:46:03.994 Status: ZJS: discovered: Slaapkamer MR/Slaapkamer MR_temperature_air (unique_id: zwavejs2mqtt_0xc3631a9c_42-49-0-Air_temperature)
2023-01-06 20:46:04.094 Status: ZJS: discovered: Slaapkamer MR/Slaapkamer MR_humidity_air (unique_id: zwavejs2mqtt_0xc3631a9c_42-49-0-Humidity)
2023-01-06 20:46:04.195 Status: ZJS: discovered: Slaapkamer MR/Slaapkamer MR_temperature_dew_point (unique_id: zwavejs2mqtt_0xc3631a9c_42-49-0-Dew_point)
2023-01-06 20:46:04.296 Status: ZJS: discovered: Slaapkamer MR/Slaapkamer MR_notification_heat_alarm_heat_sensor_status (unique_id: zwavejs2mqtt_0xc3631a9c_42-113-0-Heat_Alarm-Heat_sensor_status)
2023-01-06 20:46:04.397 Status: ZJS: discovered: Slaapkamer MR/Slaapkamer MR_battery_maintenance_status (unique_id: zwavejs2mqtt_0xc3631a9c_42-113-0-Power_Management-Battery_maintenance_status)
2023-01-06 20:46:04.498 Status: ZJS: discovered: Slaapkamer MR/Slaapkamer MR_moisture_alarm_status (unique_id: zwavejs2mqtt_0xc3631a9c_42-113-0-Weather_Alarm-Moisture_alarm_status)
2023-01-06 20:46:04.598 Status: ZJS: discovered: Slaapkamer MR/Slaapkamer MR_battery_level (unique_id: zwavejs2mqtt_0xc3631a9c_42-128-0-level)
2023-01-06 20:46:04.699 Status: ZJS: discovered: Slaapkamer MR/Slaapkamer MR_battery_islow (unique_id: zwavejs2mqtt_0xc3631a9c_42-128-0-isLow) 

When looking up the devices in Domoticz (Setup -> Devices) it does show most of the "discovered" from above but emperature_air (unique_id: zwavejs2mqtt_0xc3631a9c_42-49-0-Air_temperature) is missing. I do see this behavior at multiple ZWA039 devices.

Any workaround or solution for this?

_Originally posted by @rlouwerens in https://github.com/domoticz/domoticz/issues/5305#issuecomment-1374056762_

**Comment by gizmocuz** [comment]
When reporting mqtt-ad issues, please provide the following details:

all config topic (something like homeassistant/sensor/xyz/temperature/config
and also the contents of this (a json text object)

If you look in the config topic, you will see also references to state topics like zwavejsui/name/xx/yy
In these the actual values are presented

With this information we should be able to completely replicate your device and debug

**Comment by duck-wolv** [comment]
Thanks for the fast reply!

Not sure if I did what you asked, but I've added two files in this comment: node42_debug.log which comes from Z-Wave JS UI (export of the node) and node_42.json.log which I grapped from the ZWave JS UI Debug window while I pressed the "Awake" button at the ZWA039 device.

I verified that the "/config" topic is in the file, hopefully this gives the information required. Please let me know when more logging or details are needed.

[node42_debug.log](https://github.com/domoticz/domoticz/files/10366118/node42_debug.log)
[node_42.json.log](https://github.com/domoticz/domoticz/files/10366119/node_42.json.log)

**Comment by gizmocuz** [comment]
Sorry, I forgot to mention, please use MQTT Explorer for this

**Comment by duck-wolv** [comment]
See attached file, it contains all the config topics from the ZWA039 device copied via MQTT Explorer. I've also attached, per config topic, the value which belongs to it. 
![ZWA039_Domoticz_values](https://user-images.githubusercontent.com/1925581/211168370-c08a63fb-ad79-4911-9c09-f50ea699db23.png)

![ZWA039_zwave-js-ui_values](https://user-images.githubusercontent.com/1925581/211168334-bc74d8b1-60e7-4707-82be-1db45c91c400.png)


In your first post you mentioned that MQTT explorer should show values (temp/dew) in the topic value but as far as I can see it is not containing data. But when I look it up in Z-Wave JS UI it has, see attached screenshot for the values.


[node_42_mqtt_explorer.log](https://github.com/domoticz/domoticz/files/10367086/node_42_mqtt_explorer.log)

**Comment by waltervl** [comment]
There is a possible alternative fix by changing zwavejsui node config in forum topic https://www.domoticz.com/forum/viewtopic.php?p=297455#p297455

**Comment by gizmocuz** [comment]
Thank you for this. Could you also post the contents of the state topics:

zwave/nodeID_42/49/0/Air_temperature
zwave/nodeID_42/49/0/Humidity
zwave/nodeID_42/49/0/Dew_point

**Comment by duck-wolv** [comment]
See attached image for visuals, see lines below for the json representation:
{"time":1672904420690,"value":10.6,"nodeName":"","nodeLocation":""}
{"time":1672904420518,"value":15.1,"nodeName":"","nodeLocation":""}
{"time":1672904420579,"value":75,"nodeName":"","nodeLocation":""}

![image](https://user-images.githubusercontent.com/1925581/211372361-d1876429-7d0f-46c4-8c8c-5ba8b6860b81.png)

**Comment by waltervl** [comment]
@gizmocuz I suppose commit https://github.com/domoticz/domoticz/commit/53b6593aeb2261b4aa07de1be2aaa9f80c0f7c2c  (beta build 14930) should fix this issue?

**Comment by gizmocuz** [comment]
It's not ready yet

**Comment by gizmocuz** [comment]
@rlouwerens , is it possible you can try beta 14938?

You should get a new dew_point device and the correct temperature should now be used for the temp+hum device

**Comment by duck-wolv** [comment]
Thanks @gizmocuz , just tested and indeed, a `dew_point` device is added with the correct temperature (7.5C) and after restarting with docker-compose, removing the device in Domoticz and pressing the activate button at the ZWA039 I do now have the air temperature separated from the dew temperature. 

Although that is good news, I do see that the air temp values do not match with the values in Z-Wave JS UI (the values showed in Domoticz are 1 to 3 degrees Celsius to high and also it is not realistic the real temperature from the sensor in the room). Which data do you need to verify the this?

I also found that the ID (Setup -> Devices) is slightly different for the Temp/Humidity device: it has `_node42` as end while all others from the ZWA039 have `Node42`, for my seconds ZWA039 device I do see exactly the same pattern. Not sure if this is something that needs investigation. See attached screenshots for the exact values.

![image](https://user-images.githubusercontent.com/1925581/211625924-584f6609-8ee1-4447-af48-5183a8466fe7.png)
![image](https://user-images.githubusercontent.com/1925581/211626012-ade5c47e-54bd-4620-b71b-0990e62a18bf.png)
![image](https://user-images.githubusercontent.com/1925581/211626086-983e4a6a-5e60-4e3f-9f0c-403ba4fa9c5d.png)
![image](https://user-images.githubusercontent.com/1925581/211626231-697b3137-542a-41c1-a040-2987faeda0f2.png)

**Comment by duck-wolv** [comment]
@gizmocuz It looks like that all temperature devices (so other brands/types as well) have a mismatch between Z-wave JS UI and Domoticz.

**Comment by duck-wolv** [comment]
Right now, a few hours after my last comment and without any change, it seems that the temperature sensors are showing the correct temperature.

So with that it looks like this ticket can be closed. Many thanks for the fix!

**Comment by gizmocuz** [comment]
Thanks for the feedback.
Well, Domoticz only updates the temperature sensors when it is received.
Because you updated Domoticz, you probably missed a update and old values are shown.
Now this is not important for temperature sensors, and they get updated frequently,
but if you are depending heavily on a temperature value, you can enable the 'retain' flag for this sensor in ZWaveJS-UI

**Comment by duck-wolv** [comment]
@gizmocuz I still see some strange temperature ups and downs in my ZWA024 temperature values which I believe is wrong (ZWA039 are fine now). The sensor is located in the living room and should be around 18 degrees C and can't be fluctuating more then 0.5C max. I tried all options from Z-Wave JS UI ("Clear retained", "Update topics", "Refresh values", "Re-interview") but nothing seems to fix the ups and downs. I even removed the device from Domoticz and added it afterwards again.

Could any of this be the result of the latest change you implemented?

![image](https://user-images.githubusercontent.com/1925581/211894890-ac391c65-a345-476d-8fb4-45c4da7c36f7.png)
![image](https://user-images.githubusercontent.com/1925581/211895876-029fa3ef-98f6-4524-9145-6b6843b2ff66.png)

**Comment by waltervl** [comment]
With mqtt explorer you can check the history of the values sent by Zwavejsui to see if it is the source of your value changes.
http://mqtt-explorer.com/img/screen2.png

**Comment by duck-wolv** [comment]
@waltervl Thanks for sharing! I've tried but I can't get any history from MQTT Explorer (for any device and any topic). It only shows the last (I guess) value and not even in a graph but in json format.


![image](https://user-images.githubusercontent.com/1925581/211993137-4bc35ce5-47ff-44a4-81c5-6548d44978b2.png)
![image](https://user-images.githubusercontent.com/1925581/211994403-166f7d7b-af28-430c-9123-4b5741785acb.png)

**Comment by gizmocuz** [comment]
I do not have this issue

**Comment by gizmocuz** [comment]
Do you have more temperature sensors? Maybe it is indeed this sensor

**Comment by duck-wolv** [comment]
> Do you have more temperature sensors? Maybe it is indeed this sensor

I do have multiple sensors and from what I can see is that both Multisensor 6 (ZW100) and Multisensor 7 (ZWA024) have this issue. Other sensors seem to be fine.

Output from ZW100:
![image](https://user-images.githubusercontent.com/1925581/212325697-88960441-7102-4067-b10f-dca17a5cc032.png)

**Comment by gizmocuz** [comment]
What I see here is not a pattern that is uncommon (temperature going low, humidity up, and visa versa)

I suspect the multisensor has only one temperature and humidity field?

This is my temperature reading from a multisensor.... fluctuation of 0.1

![image](https://user-images.githubusercontent.com/2265299/212330412-a5dd4b1a-c04c-4b31-9db3-e5c4e2388799.png)

(running beta 14956)

**Comment by duck-wolv** [comment]
> I suspect the multisensor has only one temperature and humidity field?

Indeed, it has an air temp and humity field, no dew. As it is a multi sensor it has other fields as well.

![image](https://user-images.githubusercontent.com/1925581/212331450-30e650bf-c5a3-480b-883a-d8fe302ed984.png)

**Comment by gizmocuz** [comment]
Well I am very certain that my patch to fix the dew-point has nothing to do with the behavior about

Have a look at the change:

https://github.com/domoticz/domoticz/commit/53b6593aeb2261b4aa07de1be2aaa9f80c0f7c2c

Could you increase the reporting interval? (Assuming the nodes are powered)
Else it looks like they are two different temperature sensors?

**Comment by duck-wolv** [comment]
> Could you increase the reporting interval? (Assuming the nodes are powered)

That seems to work! 

Maybe I found the rootcause of the fluctuating: while checking the second Multisensor 6, which is located in another room without heating and have a temperature around 14 C, I found that it was giving exactly the same temp/humidity like the Multisensor 7. After some debugging I found that I both had given the same name in Z-Wave JS UI (Name: "Multisensor") with a different location. I guess Domoticz is confused by that or ZJS is not sending  unique topics when the name field is the same? Without looking up to code I can imagine that the device value in Domoticz is updated by both the values (1 from MS 6 and another one from MS 7), resulting in 18.x degrees one time, and 14.x another time.
